src/dmd/dsymbolsem.d

```diff
@@ -5928,6 +5928,18 @@ void templateInstanceSemantic(TemplateInstance tempinst, Scope* sc, Expressions*
     {
         tempinst.minst = null;
     }
+    version (all)
+    {
+        // https://issues.dlang.org/show_bug.cgi?id=21299
+        // This instance should come from the same instantiating module
+        // or speculative context as its enclosing template symbol.
```

Comment:
This variant looks much better. - The comment is a bit misleading though; you're only adjusting `minst` to the parent's in case this child isn't speculative (i.e., `sc.minst` is set).

Reply:
Right, I didn't want to interfere with the explicit null setting immediately before this (which has another bugzilla reference).

Reply:
That one is mutually exclusive, due to the `!tempinst.tinst` condition. - I think ignoring speculative children is fine, I'd just fix up the comment.